### DISCUSSION TITLE
Item with multiple assets & S/N?

### DISCUSSION THREAD

**Discussion by freddyboombots** [sysadminsmedia/homebox#273] [Q&A]
Wondering if anyone is assigning separate asset IDs or serial numbers for items with multiple quantities.  
ie: if a product like "2 TB SSD" has a quantity of 10, and each SSD has a different serial number, I'd like to assign a separate asset ID to each one. However, at the moment, I can only add a single asset and serial number.

Would it make more sense to create 10 separate SSD items, each with its own asset ID and serial number, or is there a better way to manage multiple quantities under one product?

Thanks.

**Comment by tankerkiller125** [comment]
Generally, the way SSDs for example would be handled would be separate asset IDs. With the parent being the machine that they're installed in, and so forth so on.

**Reply by freddyboombots** [reply]
Agreed, but I’ve been unable to create separate asset IDs for a item.  It would be helpful if additional asset IDs could be added to the item's details, similar to how a new custom line is created with the value being the serial number of the actual asset. This would allow us to manage a single item like a "2 TB SSD" while assigning separate asset IDs and tracking each serial number independently.

The ability to track multiple assets under one item would provide better control and flexibility, especially when scanning assets into parent containers (like 5 SSDs into a 5-bay NAS, 2 into a 2-bay NAS) or managing other scenarios such as selling, keeping one for hot swaps, or writing off damaged items. Even in such cases, being able to track each asset individually would be valuable.

**Reply by PAS-BC** [reply]
I have just purchased two (2) new '2 TB' 3.5" drives for my QNAP 5 bay NAS.

One HDD as replacement for a failed drive, the other as a 'cold spare' (so I don't have to wait 2 weeks when the next original drive fails LOL).

I figured it was a good opportunity to create an Item for the NAS and 6 more items for the HDDs (inc the spare).
The fact that the new drives have 3 year warrant was an extra incentive to make use of the 'Warranty' fields in HomeBox.

I created an Item for the NAS and then five (5) more items for the HDDs, with their 'Parent Item' = the 'NAS' item.
Each HDD has the Serial Number, cost, warranty etc. detail recorded.

The 'NAS' item has Location = 'Radio Shack'.
The 'Cold Spare' HDD has Location = 'Radio Shack' (because it's just in a cupboard in the Shack).

Seems a good use of 'Item' within another 'Item' - only the 'Parent Item' needs to be in a Location.

**Comment by Openuser87** [comment]
I would support the idea that some kind of grouping of identical objects is needed. E.g., in smarthome / IoT area, we usually have multiple identical items (plugs, TRVs, sensors...) which are same model, but purchased on different dates, in different shops, having different serial numbers and other parameters...

So it would be useful to have a "group" (be it implemented as an "item" or something else) showing we have 5 plugs of model X by vendor Y, consisting of 5 separate objects with their own individual parameters.

**Comment by raimund-schluessler** [comment]
I would also like to second this idea. I currently maintain my own inventory app which uses the concept of `items` and `item instances`. An `item` in that sense is a non-physical thing depicting the shared properties of all the `item instances` belonging to this `item`. These properties are e.g. the maker, model number, GTIN or manuals. But an item has no location, quantity or purchase price. Only if you actually buy an item, it gets an `instance` with assigned quantity, location, price, purchase data and invoice.

In the concept of how `HomeBox` manages objects, this could be mapped with `items` and `purchases` (as the purchase being the item instance from the example above). The required changes wouldn't be dramatic, I think:
- `Purchases` would need to have a quantity, location and attachments (i.e. the invoice), effectively allowing them to be `instances`.
- For `items` it would be enough to allow them to not have a `location`, i.e. lose their physical presence.
